From 00408d87005c0c98186d009f87c1dd4f07872738 Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Mon, 14 Aug 2023 15:12:26 +0200
Subject: [PATCH 0537/1141] irqchip/stm32-exti: Fix return from
 irq_domain_ops::select

The caller irq_find_matching_fwspec() expects select() to return
non-zero on matching irq domain.

Don't propagate the error code as it is wrongly detected as a
matching irq domain.

Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: Ie85b572354587b597a55b2fce883cd8cc81c5765
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/329517
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
Reviewed-by: Alexandre TORGUE <alexandre.torgue@foss.st.com>
---
 drivers/irqchip/irq-stm32-exti.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/irqchip/irq-stm32-exti.c b/drivers/irqchip/irq-stm32-exti.c
index 45a75e4dbcb6..46392ba7f98a 100644
--- a/drivers/irqchip/irq-stm32-exti.c
+++ b/drivers/irqchip/irq-stm32-exti.c
@@ -808,7 +808,7 @@ static int stm32_exti_h_domain_select(struct irq_domain *dm,
 
 	ret = of_irq_parse_raw(NULL, &out_irq);
 	if (ret)
-		return ret;
+		return 0;
 
 	return (dm->parent->fwnode == of_node_to_fwnode(out_irq.np));
 }
-- 
2.39.2

